include/tvm/runtime/packed_func.h

```diff
@@ -229,13 +229,13 @@ class TypedPackedFunc<R(Args...)> {
    * \endcode
    *
    * \param typed_lambda typed lambda function.
+   * \param name the name of the lambda function.
    * \tparam FLambda the type of the lambda function.
    */
-  template <typename FLambda, typename = typename std::enable_if<
-                                  std::is_convertible<FLambda,
-                                                      std::function<R(Args...)>>::value>::type>
-  TypedPackedFunc(const FLambda& typed_lambda) {  // NOLINT(*)
-    this->AssignTypedLambda(typed_lambda);
+  template <typename FLambda, typename = typename std::enable_if<std::is_convertible<
+                                  FLambda, std::function<R(Args...)>>::value>::type>
+  TypedPackedFunc(const FLambda& typed_lambda, std::string name = "<anonymous>") {  // NOLINT(*)
```

Comment:
We need two constructors, one that has the string argument and another that does not.

The one that is anonymous should not capture any string content or construct a string

Reply:
Good catch. I totally forgot about that constructor.